invenio_pages/records/models.py

```diff
@@ -39,23 +45,85 @@ class PageModel(db.Model, Timestamp):
     template_name = db.Column(db.String(70), nullable=False)
     """Page template name."""
 
+    @classmethod
+    def create(self, data):
+        """Create a new page."""
+        try:
+            with db.session.begin_nested():
+                obj = PageModel(
+                    url=data["url"],
+                    title=data.get("title", ""),
+                    content=data.get("content", ""),
+                    description=data.get("description", ""),
+                    template_name=data["template_name"],
+                )
+                db.session.add(obj)
+
+            return obj
+        except IntegrityError:
+            return None
+
     @classmethod
     def get_by_url(self, url):
         """Get a page by URL.
 
         :param url: The page URL.
         :returns: A :class:`invenio_pages.records.models.PageModel` instance.
         """
-        return PageModel.query.filter_by(url=url).one()
+        try:
+            return self.query.filter_by(url=url).one()
+        except NoResultFound:
+            raise PageNotFoundError(url)
 
     @classmethod
-    def get_by_id(self, id):
+    def get(self, id):
         """Get a page by ID.
 
         :param id: The page ID.
         :returns: A :class:`invenio_pages.records.models.PageModel` instance.
         """
-        return PageModel.query.filter_by(id=id).one()
+        try:
+            return self.query.filter_by(id=id).one()
+        except NoResultFound:
+            raise PageNotFoundError(id)
+
+    @classmethod
+    def search(self, search_params={}, filters=[]):
```

Comment:
dictionary and list are mutable data collections, it is not a good practice to set them as default value of methods' arguments